service/framework_test.go

```diff
@@ -65,6 +65,15 @@ func Test_Service_NewVersionedResources(t *testing.T) {
 					"servicev2",
 					"endpointsv2",
 				},
+				"2.1.0": []string{
```

Comment:
Why is this now a minor bump? AFAIK you fix a bug, which would be a patch which we want to roll automatically in the future. So this difference is quite important. Maybe I miss something.

Reply:
Good to check this. This week we also need to make Azure AZ / OIDC enabled in some installations. That is a bigger change so having a `2.1.0` release for both makes sense to me.

https://github.com/giantswarm/giantswarm/issues/2413

IMO would be good to post about this is `#sig-operator` since versioning means we need to do more communication.

Reply:
I don't follow. This here is a version bundle version. This is different than a release version. We should also not try to keep release versions aligned between providers. It will just not work out. We will have different versions and development cycles across providers and this is fine. 

I also do not see the relation to Azure here otherwise. This is AWS and fixing a bug means a patch release for the version bundle.

Reply:
Damnit. I had a typo in my comment its `Azure AD / OIDC`. The customer wants to use Azure Active Directory for authentication with their AWS guest clusters. This is the story @puja108 has been working on.

We need to create a `2.1.0` version bundle before they move their production clusters. We could create both `2.0.1` and `2.1.0` version bundles this week but IMO this is unnecessary overhead.

Reply:
Fair Ross, but this is not of interest here. Here we only fix a bug and thus should create a patch release for the version bundle. I do not see the overhead. This overhead is something we have to do continuously when fixing bugs. When there is overhead we should improve the process and our tooling, but not work around either. IMO this here should be `2.0.1`. When we have to create another version tomorrow, so be it. Then we use the current state, which is then `2.0.1` and create `2.1.0` based on it.

Reply:
OK we can deprecate `2.0.0` once this is ready. Makes sense to go with `2.0.1`  and put the update process into practice.

Reply:
Agreed. Deprecating is cool then. Finally getting traction on the process is good. Once we live the process lets get straight to the pain points so we can identify and eliminate them with improvements on the process and our tooling.

Reply:
Ok, added `2.0.1` as WIP, PTAL.